x-pack/metricbeat/module/aws/cloudwatch/cloudwatch.go

```diff
@@ -189,14 +202,51 @@ func (m *MetricSet) Fetch(report mb.ReporterV2) error {
 
 		for namespace, namespaceDetails := range namespaceDetailTotal {
 			m.logger.Debugf("Collected metrics from namespace %s", namespace)
-
 			// filter listMetricsOutput by detailed configuration per each namespace
 			filteredMetricWithStatsTotal := filterListMetricsOutput(listMetricsOutput, namespace, namespaceDetails)
 
 			// get resource type filters and tags filters for each namespace
 			resourceTypeTagFilters := constructTagsFilters(namespaceDetails)
 
-			eventsWithIdentifier, err := m.createEvents(svcCloudwatch, svcResourceAPI, filteredMetricWithStatsTotal, resourceTypeTagFilters, regionName, startTime, endTime)
+			//Check whether namespace is APIGW
+			var (
+				checkns_lower            = strings.ToLower(checkns)
+				checkresource_type_lower = strings.ToLower(checkresource_type)
+			)
+			useonlyrest := false
+			if len(resourceTypeTagFilters) == 1 {
+				for key := range resourceTypeTagFilters {
+					if strings.Compare(strings.ToLower(key), checkresource_type_lower) == 0 {
+						useonlyrest = true
+					}
+				}
+			}
+			if strings.Contains(strings.ToLower(namespace), checkns_lower) {
+				// inforestapi includes only Rest APIs
+				if useonlyrest {
+					infoapi, err = aws.GetAPIGatewayRestAPIOutput(APIClients.Apigateway, config.LimitRestAPI)
+					if err != nil {
+						m.Logger().Errorf("could not get rest apis output: %v", err)
+					}
+				} else {
+					infoapi, err = aws.GetAPIGatewayRestAPIOutput(APIClients.Apigateway, config.LimitRestAPI)
+					if err != nil {
+						m.Logger().Errorf("could not get rest apis output: %v", err)
+					}
+					// infoapi includes only WebSocket and HTTP APIs
+					infotherapi, err := aws.GetAPIGatewayAPIOutput(APIClients.Apigatewayv2)
+					if err != nil {
+						m.Logger().Errorf("could not get http and websocket apis output: %v", err)
+					}
+					if len(infotherapi) > 0 {
+						maps.Copy(infoapi, infotherapi)
+					}
+				}
+
+				m.Logger().Infof("infoapi response: %v", infoapi)
```

Comment:
should this be info level or debug level?

Reply:
Yes done in [535b905](https://github.com/elastic/beats/pull/40755/commits/535b9059fa11776e454bfd89742c7e9cdd78b555)